akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolConductor.scala

```diff
@@ -140,15 +140,15 @@ private object PoolConductor {
               slotStates(slotIx) = slotStateAfterDisconnect(slotStates(slotIx), failed)
           }
           pull(slotIn)
-          val tryPull = nextSlot == -1
+          val wasBlocked = nextSlot == -1
           nextSlot = bestSlot()
-          if (tryPull) tryPullCtx()
+          if (wasBlocked && nextSlot != -1) pull(ctxIn)
```

Comment:
I am still not fully understanding what this intends to do. I get it that it checks for the transition of nextSlot == -1 to nextSlot != -1, but I don't get why this is a useful event to decide when to pull `ctxIn`?